src/Controls/src/Core/Handlers/Items/iOS/ItemsViewLayout.cs

```diff
@@ -595,5 +604,51 @@ internal void ClearCellSizeCache()
 		{
 			_cellSizeCache.Clear();
 		}
+
+		CGSize TryFindEstimatedSize(CGSize existingMeasurement)
+		{
+			if (CollectionView == null || GetPrototypeForIndexPath == null)
+				return existingMeasurement;
+
+			//Since this issue only seems to be reproducible on Horizontal scrolling, we only check for that
+			if (ScrollDirection == UICollectionViewScrollDirection.Horizontal)
+			{
+				return FindEstimatedSizeUsingWidth(existingMeasurement);
+			}
+
+			return existingMeasurement;
+		}
+
+		CGSize FindEstimatedSizeUsingWidth(CGSize existingMeasurement)
+		{
+			// TODO: Handle grouping
+			var group = 0;
+			var collectionViewWidth = CollectionView.Bounds.Width;
+			var numberOfItemsInGroup = CollectionView.NumberOfItemsInSection(group);
+			
+			// Calculate the number of cells that can fit in the viewport
+			var numberOfCellsToCheck = Math.Min((int)(collectionViewWidth / existingMeasurement.Width) + 1, numberOfItemsInGroup);
+			
+			// Iterate through the cells and find the one with a wider width
+			for (int i = 1; i < numberOfCellsToCheck; i++)
+			{
+				var indexPath = NSIndexPath.Create(group, i);
+				if (GetPrototypeForIndexPath(indexPath) is ItemsViewCell cellAtIndex)
+				{
+					cellAtIndex.ConstrainTo(ConstrainedDimension);
+					var measureCellAtIndex = cellAtIndex.Measure();
+					
+					// Check if the cell has a wider width
+					if (measureCellAtIndex.Width > existingMeasurement.Width)
```

Comment:
If we are just looking at the widest width, will the scrollbar make the list appear larger but as we scroll it will "shrink"? More checking to see we don't end up with a white space. Not sure how strict the collection view is to preserve/ensure this width is adhered to.